misk-datadog/src/main/kotlin/misk/tracing/backends/datadog/DatadogTracingBackendModule.kt

```diff
@@ -1,19 +1,22 @@
 package misk.tracing.backends.datadog
 
-import datadog.trace.api.GlobalTracer
-import misk.inject.KAbstractModule
+import datadog.opentracing.DDTracer
 import io.opentracing.Tracer
 import io.opentracing.noop.NoopTracerFactory
+import io.opentracing.util.GlobalTracer
+import misk.inject.KAbstractModule
 import misk.logging.getLogger
 
 /**
  * Binds the datadog tracer to opentracing's [Tracer]
  */
 class DatadogTracingBackendModule : KAbstractModule() {
   override fun configure() {
-    // A DDTracer is installed by the DataDog Java agent, which runs before the app's main() method.
-    val tracer = GlobalTracer.get()
-    if (tracer is Tracer) {
+    // N.B. For information on the registration of the DDTracer w/ OpenTracing,
+    // see https://docs.datadoghq.com/tracing/custom_instrumentation/java/
+    val tracer = DDTracer.builder().build()
+    if (GlobalTracer.registerIfAbsent(tracer)) {
```

Comment:
when does this return true/false?

Reply:
If there is already a tracer registered to the OpenTracing global tracer. Shouldn't happen unless someone else went and registered some other tracer.